### PR TITLE
Random spell flags

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Features "Adds RANDOM_ spell flags for duration, damage, and aoe"```
#### Purpose of change

Some of the AEA "spells" were made without C++ support for their variable duration. This adds support for the duration that's used, and adds a few more for use.
#### Describe the solution

the RANDOM_ flags will, instead of choosing the value based on your spell level alone, will pick a value between the minimum (which increases by level) and the maximum using the `rng` function. Additionally, the callback that prints the spellcasting info was updated to show this properly.

### CODE REVIEW

File: src/magic.cpp [hunk 1]
```diff
@@ -175,6 +179,15 @@ static damage_type damage_type_from_string( const std::string &str )
     }
 }
 
+static std::string moves_to_string( const int moves )
```
Comment (BevapDin) [inline src/magic.cpp]: There is already an equal version of this function defined in "magic.cpp". Do we really need another copy if it here?

File: src/magic.cpp [hunk 2]
```diff
@@ -405,23 +418,76 @@ int spell::field_intensity() const
...
+        }
     }
 }
 
+std::string spell::damage_string() const
+{
+    if( has_flag( spell_flag::RANDOM_DAMAGE ) ) {
+        return string_format( "%d-%d", min_leveled_damage(), type->max_damage );
+    } else {
+        const int dmg = damage();
```
Comment (BevapDin) [inline src/magic.cpp]: Do you really need a variable when you only use it *once* anyway? (You call `damage()` below again.)

File: src/magic.cpp [hunk 3]
```diff
@@ -405,23 +418,76 @@ int spell::field_intensity() const
...
+        if( dmg >= 0 ) {
+            return string_format( "%d", damage() );
+        } else {
+            return string_format( "+%d", damage() );
+        }
```
Comment (Qrox) [inline src/magic.cpp]: I think this can be simply
```
return string_format( "%+d", dmg );
```

Comment (Qrox) [reply src/magic.cpp]: I mean the two `string_format` calls can be merged into one using the `%+d` specifier. Also the current if condition is incorrect, it currently output negative numbers with a leading plus sign.

Comment (KorGgenT) [reply src/magic.cpp]: but i don't want the + in the other statement.

Comment (KorGgenT) [reply src/magic.cpp]: oh shoot, yeah. i want abs()

**Pull request merged**